### PR TITLE
Improve images

### PR DESCRIPTION
**Pull request by barrachri**
Close this issue https://github.com/aio-libs/aiodocker/issues/3
and https://github.com/aio-libs/aiodocker/issues/33

I also changed all f'' to "".format() inside images.py

### CODE REVIEW

File: aiodocker/images.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    async def history(self, name):
+        response = await self.docker._query_json(
+            "images/{name}/history".format(name=name),
+            headers={"content-type": "application/json", },
+        )
+        return response
+
+    async def pull(self,
+                   from_image: str,
+                   repo: Optional[str]=None,
```
Comment (asvetlov) [inline aiodocker/images.py]: I suggest use keyword-only arguments in every place where we have optional parameters.
The project has no many adopters yet thus we can allow this breaking change.

The signature should be `async def pull(self, from_image, *, repo=None, tag=None, stream=None)` (I've skipped type annotations).

I suggest applying this recommendation to every public API method in the library.

@achimnol what do you think about?

File: aiodocker/images.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        List of images
+        """
+        response = await self.docker._query_json(
+            "images/json", "GET",
+            params=params,
+            headers={"content-type": "application/json", },
+        )
+        return response
+
+    async def get(self, name):
```
Comment (asvetlov) [inline aiodocker/images.py]: Annotate `name`

File: aiodocker/images.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        Args:
+            name: name of the image
+        """
+        response = await self.docker._query_json(
+            "images/{name}/json".format(name=name),
+            headers={"content-type": "application/json", },
+        )
+        return response
+
+    async def history(self, name):
```
Comment (asvetlov) [inline aiodocker/images.py]: Add type annotation

File: aiodocker/images.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+        response = await self.docker._query(
+            "images/create",
+            "POST",
+            params=params,
+            headers={"content-type": "application/json", },
+        )
+        return (await json_stream_result(response, stream=stream))
+
+    async def push(self, name, tag=None, auth=None, stream=False):
```
Comment (asvetlov) [inline aiodocker/images.py]: `..., *, tag=None, ...`

Comment (asvetlov) [inline aiodocker/images.py]: Type annotations are missing

File: aiodocker/images.py [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+            params['tag'] = tag
+        response = await self.docker._query(
+            "images/{name}/push".format(name=name),
+            "POST",
+            params=params,
+            headers=headers,
+        )
+        return (await json_stream_result(response, stream=stream))
+
+    async def tag(self, name: str, repo: str, tag: Optional[str]=None) -> bool:
```
Comment (asvetlov) [inline aiodocker/images.py]: Make `tag` keyword-only

File: aiodocker/images.py [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+
+        await self.docker._query(
+            "images/{name}/tag".format(name=name),
+            "POST",
+            params=params,
+            headers={"content-type": "application/json"},
+        )
+        return True
+
+    async def delete(self, name: str, force: bool=False,
```
Comment (asvetlov) [inline aiodocker/images.py]: Make `force` and `noprune` keyword only

File: aiodocker/images.py [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+        """
+        params = {'force': force, 'noprune': noprune}
+        response = await self.docker._query_json(
+            "images/{name}".format(name=name),
+            "DELETE",
+            params=params,
+        )
+        return response
+
+    async def build(self,
```
Comment (asvetlov) [inline aiodocker/images.py]: What the behavior for calling `build()` without parameters?

Comment (asvetlov) [inline aiodocker/images.py]: All optional args should be keyword-only

Comment (barrachri) [inline aiodocker/images.py]: We should probably raise an error if neither remote or fileobj are specified.

Comment (asvetlov) [inline aiodocker/images.py]: Sounds good

File: aiodocker/images.py (hunk 1 above)
Comment (achimnol) [inline aiodocker/images.py]: Yeah, I agree with that and can bear with this breaking changes in my projects.
It's good to have consistent styles/conventions with other aio-libs projects, particularly, aiohttp.

**Comment by barrachri** [comment]
I think only history was missing.
I added the test and did a rebase, I guess is ready to be merged.

**Comment by asvetlov** [comment]
Ok, LGTM.
@achimnol @jettify would you review as well and merge?
Otherwise I'll do it later. After tomorrow maybe if will not give your feedbacks.

File: aiodocker/utils.py [hunk 8]
```diff
@@ -57,3 +63,89 @@ def httpize(d: Optional[dict]) -> Optional[dict]:
...
+def format_env(key, value):
+    """
+    Format envs from {key:value} to ['key=value']
+    """
+    if value is None:
+        return key
+    if isinstance(value, bytes):
+        value = value.decode('utf-8')
+
+    return f"{key}={value}"
```
Comment (jettify) [inline aiodocker/utils.py]: should be moved to format

File: aiodocker/utils.py [hunk 9]
```diff
@@ -57,3 +63,89 @@ def httpize(d: Optional[dict]) -> Optional[dict]:
...
+        dfinfo = tarfile.TarInfo('Dockerfile')
+        dfinfo.size = len(fileobject.getvalue())
+        fileobject.seek(0)
+    else:
+        dfinfo = t.gettarinfo(fileobj=fileobject, arcname='Dockerfile')
+
+    t.addfile(dfinfo, fileobject)
+    t.close()
+    f.seek(0)
+    return f
```
Comment (jettify) [inline aiodocker/utils.py]: lets add new line

**Comment by jettify** [comment]
I think we can merge, things I would like to see addressed in next PRs:

1) we are adding type annotation, to make use of them we need to add mypy checks
2) we add `headers={"content-type": "application/json"},` everywhere
3) we definitely need to add `pep8` checker and forget about style errors forever, I added pyflakes and have plan for pep8, but if someone wish to do this, please do.

**Comment by asvetlov** [comment]
@jettify thanks! Four eyes are better than two ones :)

**Comment by barrachri** [comment]
I guess I have to rebase and send and PR.

**Comment by asvetlov** [comment]
You could just merge master -- we do squash anyway.

**Comment by asvetlov** [comment]
Perfect! Thanks!

**Pull request merged**